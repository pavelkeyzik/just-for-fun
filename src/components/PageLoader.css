.PageLoader__spinner-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  padding: 40px;
}
