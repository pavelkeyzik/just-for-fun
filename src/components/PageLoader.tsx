import React from 'react';
import { Spinner, Container, Row, Col } from 'reactstrap';

import styles from './PageLoader.module.css';

export interface IPageLoaderProps {
  message: string;
}

export function PageLoader({ message }: IPageLoaderProps): JSX.Element {
  return (
    <Container>
      <Row>
        <Col>
          <div className={styles.loader}>
            <Spinner type="grow" color="success" />
            <b>{message}</b>
          </div>
        </Col>
      </Row>
    </Container>
  );
}
