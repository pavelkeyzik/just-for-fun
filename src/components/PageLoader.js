import React from 'react';
import { Spinner, Container, Row, Col } from 'reactstrap';

import './PageLoader.css';

export function PageLoader({ message }) {
  return (
    <Container>
      <Row>
        <Col>
          <div className="PageLoader__spinner-container">
            <Spinner type="grow" color="success" />
            <b>{message}</b>
          </div>
        </Col>
      </Row>
    </Container>
  );
}
