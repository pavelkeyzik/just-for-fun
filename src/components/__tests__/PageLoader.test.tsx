import React from 'react';
import { render } from '@testing-library/react';
import { PageLoader } from '../PageLoader';

test('Should return loader with message `Hello World`', () => {
  const message = 'Hello World';
  const { getAllByText } = render(<PageLoader message={message} />);

  const elementsWithText = getAllByText(message);

  expect(elementsWithText).toHaveLength(1);
});

test('Check snapshot with `Hello World`', () => {
  const message = 'Hello World';
  const { container } = render(<PageLoader message={message} />);

  expect(container.firstChild).toMatchSnapshot();
});
