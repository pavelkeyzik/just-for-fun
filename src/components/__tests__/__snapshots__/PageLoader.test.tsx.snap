// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot with \`Hello World\` 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="loader"
      >
        <div
          class="spinner-grow text-success"
          role="status"
        >
          <span
            class="sr-only"
          >
            Loading...
          </span>
        </div>
        <b>
          Hello World
        </b>
      </div>
    </div>
  </div>
</div>
`;
